code/datums/components/archaeology.dm

```diff
@@ -0,0 +1,97 @@
+/* How this works:
+		In turf.Initialize() , AddComponent(/datum/component/archaeology, prob2drop). prob2drop is a base number that affects each drop in list/drop the same. Good for if you want to have turfs be a bit more dynamic.
+		In the turf's vars, you can set the drop list with the archdrops var. Format is type = num where num is the max it could possibly drop. postdig
+		In the turf's AttackBy() [open turfs are vastly different and don't typically call inheritance hence component], call ArchaeologySignal(user, W). This will send the signal to the component as well.
+		If your turf has a unique post-dig sprite like basalt/asteroid, put that in postdig_icon as a text string AND set postdig_icon_change = TRUE.
+*/
+/datum/component/archaeology
+	dupe_type = COMPONENT_DUPE_UNIQUE
+	var/list/drops = list()
+	var/prob2drop = 0
+	var/mob/user
+	var/obj/item/W
+	var/dug = FALSE
+
+/datum/component/archaeology/Initialize(_prob2drop)
+	prob2drop = Clamp(_prob2drop, 0, 100)
+	if(isopenturf(parent))
+		var/turf/open/OT = parent
+		drops = OT.archdrops
```

Comment:
This should be being passed as an argument to AddComponent, not being looked up manually

Reply:
if it's turf specific how would I make this without being one of those 200 switch memes

Reply:
see his comment beloooooow~

Reply:
so the issue is not archdrops but rather how the component obtains archdrops in Initialize instead of AddComponent?

Reply:
Let me just show you what I think it is you wanted me to do and we'll just go from there because it's not 100% clicking.